Fix systemd in chroot

kwohlfahrt opened the pull request:
The 'request ignored' message is in stderr, not stdout.

##### SUMMARY
In a chroot environment, `systemd show` prints `Running in chroot, request ignored` to stderr. However, the module checks stdout, and therefore never catches this condition and instead reports `Could not find the requested service` when a systemd service is enabled in a chroot.

Updates the original fix for #21026 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
systemd module

##### ANSIBLE VERSION
```
ansible 2.4.0
```

Commit:
Fix systemd in chroot

The 'request ignored' message is in stderr, not stdout.

bcoca commented on lib/ansible/modules/system/systemd.py:
```diff
@@ -307,7 +307,7 @@ def main():
     (rc, out, err) = module.run_command("%s show '%s'" % (systemctl, unit))
 
-    if out.find('ignoring request') != -1:
+    if err.find('ignoring request') != -1:
```
we should search both

kwohlfahrt replied:
Could you explain the need to check stdout? `systemctl show <service>` dumps a lot of information about the service being changed. It might be unusual, but the`Description` field for example could contain the text 'ignoring request'.

bcoca replied:
because the version i tested with initially had the message in stdout

kwohlfahrt replied:
Fair. I've attempted to avoid false positives by making sure '=' is not in the output in the updated version.

Commit:
Check both stdout and stderr for systemd message

Some versions of systemd report to stderr, others to stdout. Also check
whether output could be a valid normal response to avoid false positives.
